x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/SpatialContains.java

```diff
@@ -87,21 +98,37 @@ protected boolean geometryRelatesGeometry(BytesRef left, BytesRef right) throws
             return geometryRelatesGeometries(left, rightComponent2Ds);
         }
 
+        private boolean geometryRelatesGeometries(MultiValuesBytesRef left, MultiValuesBytesRef right) throws IOException {
+            Component2D[] rightComponent2Ds = asLuceneComponent2Ds(crsType, right.combined());
+            return geometryRelatesGeometries(left, rightComponent2Ds);
+        }
+
         private boolean geometryRelatesGeometries(BytesRef left, Component2D[] rightComponent2Ds) throws IOException {
             Geometry leftGeom = fromBytesRef(left);
             GeometryDocValueReader leftDocValueReader = asGeometryDocValueReader(coordinateEncoder, shapeIndexer, leftGeom);
             return geometryRelatesGeometries(leftDocValueReader, rightComponent2Ds);
         }
 
+        private boolean geometryRelatesGeometries(MultiValuesBytesRef left, Component2D[] rightComponent2Ds) throws IOException {
+            GeometryDocValueReader leftDocValueReader = asGeometryDocValueReader(coordinateEncoder, shapeIndexer, left.combined());
+            return geometryRelatesGeometries(leftDocValueReader, rightComponent2Ds);
+        }
+
         private boolean geometryRelatesGeometries(GeometryDocValueReader leftDocValueReader, Component2D[] rightComponent2Ds)
             throws IOException {
+            boolean contains = true;
+            boolean intersectsButNotContains = false;
             for (Component2D rightComponent2D : rightComponent2Ds) {
+                boolean containsComp = geometryRelatesGeometry(leftDocValueReader, rightComponent2D);
```

Comment:
I don't think this is going to work. We should try to do exactly the same thing we are doing in lucene. Therefore we should create a doc value with all the geometries and then check the spatial relationship. Otherwise there can be several edge cases which are going to be a pain.

Reply:
I have simplified this now.